packages/manager/apps/pci-kubernetes/src/pages/new/view-models/selectEstimationPriceFromPlan.spec.ts

```diff
@@ -0,0 +1,143 @@
+      plan: TClusterPlanEnum.FREE,
+      pools: [] as NodePoolPrice[],
+      options: { has3AZ: true, showSavingPlan: false },
+      expected: {
+        cluster: 'kube_common_estimation_price_free',
+        node: '0 €/mo',
+        total: '0 €/mo',
+        showSavingPlan: false,
+      },
+    },
+    {
+      description: 'FREE plan with has3AZ = false',
+      plan: TClusterPlanEnum.FREE,
+      pools: [] as NodePoolPrice[],
+      options: { has3AZ: false, showSavingPlan: true },
+      expected: {
+        cluster: 'kube_common_estimation_price_free',
+        node: '0 €/mo',
+        total: '0 €/mo',
+        showSavingPlan: true,
+      },
+    },
+    {
+      description: 'FREE plan with has3AZ = false showSavingPlan = false',
+      plan: TClusterPlanEnum.FREE,
+      pools: [] as NodePoolPrice[],
+      options: { has3AZ: false, showSavingPlan: false },
+      expected: {
+        cluster: 'kube_common_estimation_price_free',
+        node: '0 €/mo',
+        total: '0 €/mo',
+        showSavingPlan: false,
+      },
+    },
+    {
+      description: 'STANDARD plan with has3AZ = false SavingPlan = true',
+      plan: TClusterPlanEnum.STANDARD,
+      pools: nodePools,
+      options: { has3AZ: false, showSavingPlan: false },
+      expected: {
+        cluster: '720 €/mo',
+        node: '300 €/mo',
+        total: '1020 €/mo',
+        showSavingPlan: false,
+      },
+    },
+    {
+      description: 'STANDARD plan with showSavingPlan = true',
+      plan: TClusterPlanEnum.STANDARD,
+      pools: nodePools,
+      options: { has3AZ: true, showSavingPlan: true },
+      expected: {
+        cluster: '720 €/mo',
+        node: '300 €/mo',
+        total: '1020 €/mo',
+        showSavingPlan: true,
+      },
+    },
+    {
+      description: 'STANDARD plan with showSavingPlan = true',
+      plan: TClusterPlanEnum.STANDARD,
+      pools: nodePools,
+      options: { has3AZ: false, showSavingPlan: true },
+      expected: {
+        cluster: '720 €/mo',
+        node: '300 €/mo',
+        total: '1020 €/mo',
+        showSavingPlan: true,
+      },
+    },
+    {
+      description: 'STANDARD plan with showSavingPlan = true',
+      plan: TClusterPlanEnum.STANDARD,
+      pools: nodePools,
+      options: { has3AZ: true, showSavingPlan: false },
+      expected: {
+        cluster: '720 €/mo',
+        node: '300 €/mo',
+        total: '1020 €/mo',
+        showSavingPlan: false,
+      },
+    },
+  ])(
+    '$description | has3AZ=$options.has3AZ | showSavingPlan=$options.showSavingPlan',
+    ({ plan, pools, options, expected }) => {
+      const { result } = renderHook(() =>
+        selectEstimationPriceFromPlans(
+          t,
+          getFormattedMonthlyCatalogPrice,
+          convertHourlyPriceToMonthly,
+        )(plan, plans, pools, options),
+      );
+
+      const rows = result.current;
+      expect(rows[0]?.show).toBe(expected.showSavingPlan);
```

Comment:
Suggestion : you should add a check for row length first (.toHaveLength), so then you can avoir putting '?.' everywhere